backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/subscriptions/SubscriptionSelectionEventDataFetcher.java

```diff
@@ -0,0 +1,84 @@
+ * Copyright (c) 2021 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.graphql.datafetchers.subscriptions;
+
+import java.util.Objects;
+
+import org.eclipse.sirius.web.annotations.graphql.GraphQLSubscriptionTypes;
+import org.eclipse.sirius.web.annotations.spring.graphql.SubscriptionDataFetcher;
+import org.eclipse.sirius.web.collaborative.api.dto.SubscribersUpdatedEventPayload;
+import org.eclipse.sirius.web.collaborative.api.services.IEditingContextEventProcessorRegistry;
+import org.eclipse.sirius.web.collaborative.selection.api.ISelectionEventProcessor;
+import org.eclipse.sirius.web.collaborative.selection.api.SelectionConfiguration;
+import org.eclipse.sirius.web.collaborative.selection.api.dto.SelectionEventInput;
+import org.eclipse.sirius.web.collaborative.selection.api.dto.SelectionRefreshedEventPayload;
+import org.eclipse.sirius.web.core.api.IPayload;
+import org.eclipse.sirius.web.graphql.datafetchers.IDataFetchingEnvironmentService;
+import org.eclipse.sirius.web.graphql.schema.SubscriptionTypeProvider;
+import org.eclipse.sirius.web.spring.graphql.api.IDataFetcherWithFieldCoordinates;
+import org.reactivestreams.Publisher;
+
+import graphql.schema.DataFetchingEnvironment;
+import reactor.core.publisher.Flux;
+
+/**
+ * The data fetcher used to send the refreshed tree to a subscription.
+ * <p>
+ * It will be used to fetch the data for the following GraphQL field:
+ * </p>
+ *
+ * <pre>
+ * type Subscription {
+ *   selectionEvent(input: SelectionEventInput): SelectionEventPayload
+ * }
+ * </pre>
+ *
+ * @author arichard
+ */
+// @formatter:off
+@GraphQLSubscriptionTypes(
+    input = SelectionEventInput.class,
+    payloads = {
+        SelectionRefreshedEventPayload.class,
+        SubscribersUpdatedEventPayload.class,
+    }
+)
+@SubscriptionDataFetcher(type = SubscriptionTypeProvider.TYPE, field = SubscriptionSelectionEventDataFetcher.SELECTION_EVENT_FIELD)
+// @formatter:on
+public class SubscriptionSelectionEventDataFetcher implements IDataFetcherWithFieldCoordinates<Publisher<IPayload>> {
+
+    public static final String SELECTION_EVENT_FIELD = "selectionEvent"; //$NON-NLS-1$
+
+    private final IDataFetchingEnvironmentService dataFetchingEnvironmentService;
+
+    private final IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry;
+
+    public SubscriptionSelectionEventDataFetcher(IDataFetchingEnvironmentService dataFetchingEnvironmentService, IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry) {
```

Comment:
Since 21ecbef06141f60ea5dd3501bb9ba3e42adb7094 `IDataFetchingEnvironmentService` may not be available in `SubscriptionEventDataFetcher`s